File: app/src/main/java/vn/loitp/app/activity/customviews/actionbar/navigationview/NavigationViewActivity.java
package vn.loitp.app.activity.customviews.actionbar.navigationview;

import android.graphics.Color;
import android.os.Bundle;
import android.util.TypedValue;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import java.util.ArrayList;
import java.util.List;

import loitp.basemaster.R;
import vn.loitp.core.base.BaseFontActivity;
import vn.loitp.core.utilities.LLog;
import vn.loitp.core.utilities.LUIUtil;
import vn.loitp.views.navigationview.LNavigationView;

public class NavigationViewActivity extends BaseFontActivity {
    private LNavigationView nv;
    private TextView tvMsg;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        nv = findViewById(R.id.nv);
        tvMsg = findViewById(R.id.tv_msg);

        nv.setColorOn(ContextCompat.getColor(activity, R.color.Red));
        nv.setColorOff(ContextCompat.getColor(activity, R.color.Gray));
        nv.getTv().setTextColor(Color.BLACK);
        LUIUtil.setTextSize(nv.getTv(), TypedValue.COMPLEX_UNIT_DIP, 20);//20dp

        final List<String> stringList = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            stringList.add("Item " + i);
        }

        nv.setStringList(stringList);
        nv.setNVCallback((index, s) -> {
            LLog.d(TAG, "onIndexChange " + index + " -> " + s);
            tvMsg.setText(index + " -> " + s);
        });
        findViewById(R.id.bt_0).setOnClickListener(view -> nv.setCurrenIndex(0));
        findViewById(R.id.bt_1).setOnClickListener(view -> nv.setCurrenIndex(stringList.size() - 1));
        findViewById(R.id.bt_2).setOnClickListener(view -> nv.setCurrenIndex(2));
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_navigation_view;
    }

}

File: app/src/main/java/vn/loitp/app/activity/customviews/actionbar/navigationview/NavigationViewActivity.kt
package vn.loitp.app.activity.customviews.actionbar.navigationview

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Bundle
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import com.loitpcore.core.utilities.LAppResource
import com.loitpcore.core.utilities.LUIUtil
import com.loitpcore.views.navigationview.LNavigationView
import kotlinx.android.synthetic.main.activity_navigation_view.*
import vn.loitp.app.R

@LogTag("NavigationViewActivity")
@IsFullScreen(false)
class NavigationViewActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_navigation_view
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        nv.apply {
            colorOn = LAppResource.getColor(R.color.red)
            colorOff = LAppResource.getColor(R.color.gray)
            tv?.setTextColor(Color.BLACK)
            LUIUtil.setTextSize(
                textView = this.tv,
                size = resources.getDimension(R.dimen.txt_medium)
            )
        }

        val stringList = ArrayList<String>()
        for (i in 0..9) {
            stringList.add("Item $i")
        }

        nv.setStringList(stringList)
        nv.setNVCallback(
            nvCallback = object : LNavigationView.NVCallback {
                @SuppressLint("SetTextI18n")
                override fun onIndexChange(index: Int, s: String?) {
                    logD("onIndexChange $index -> $s")
                    tvMsg?.text = "$index -> $s"
                }
            }
        )
        bt0.setOnClickListener { nv.setCurrenIndex(0) }
        bt1.setOnClickListener { nv.setCurrenIndex(stringList.size - 1) }
        bt2.setOnClickListener { nv.setCurrenIndex(2) }
    }
}
